### ISSUE TITLE
MacOSX installer security exception

### ISSUE THREAD

**Issue by christo4ferris** [cloudfoundry/cli#435]
The MacOSX installer can trigger a security exception if the user has their security settings for app downloads at either "Mac App Store" or "Mac App Store and identified developers". We should look into reducing the barrier by registering with Apple. @chipchilders is this something we can do at the Foundation level to get a certificate to sign the installers/binaries?

**Comment by goehmen** [comment]
@christo4ferris @chipchilders There are a lot of areas for improvement for all installers (mac osx, linux, windows).  There are many stories in my icebox today (search label:installer).  I want to get through the help refactor work, support the move to cc API 3.0 and the user security work (pwd expiration, inactivity-based session timeout, RBAC maturation, etc.) and then we could look at focusing on an installer emphasis.  Or we can talk about re-prioritization or augmenting the team. Given that I already have a story for your request, Chris, (https://www.pivotaltracker.com/story/show/66469062) I'm going to close the story that this issue opened.  We can continue discussion here or move to where ever appropriate.

**Comment by christo4ferris** [comment]
@goehmen , I'm glad that theres a story, thanks. At the very least, the README should warn the user that they may encounter the security exception if they have elevated their settings as I noted, and explain to them how to resolve it. Would that be possible? Would you prefer a PR?

**Comment by goehmen** [comment]
good point.  either way...PR is great.  or we can do it.

Greg Oehmen
Cloud Foundry Product Manager
415.205.6596

**Comment by jchesterpivotal** [comment]
Hi guys, I hate to be a +1, but ... I hate jumping through Apple's arbitrary hoops any more than necessary.

so. um.

+1

**Comment by andymoe** [comment]
**Summary:** My first ever interaction with cloud foundry was a negative one because the CLI installer on OS X is not properly signed. So... Bump.

Hi all, I'm a New-ish Pivotal employee @ Labs in SF. I signed up for a personal run.gopivotal.io account to learn about cloud foundry and maybe do some building in my free time. I have to say that my **very first** interaction with the product was trying to install the command line utility and **FAILING** to do so because of the security prompt on OS X. 

This made me an **unhappy user/customer** as I grudgingly worked around the security prompt to install the CLI on OS X. After this first interaction with the product I did not feel like exploring more and went to bed. (A week later I'm playing with cloud foundry again and have successfully tracked down this existing bug)

For all our reference here is the Apple guide on signing packages for OS X:
https://developer.apple.com/library/mac/documentation/IDEs/Conceptual/AppDistributionGuide/DistributingApplicationsOutside/DistributingApplicationsOutside.html#//apple_ref/doc/uid/TP40012582-CH12-SW2

When I'm feeling industrious some other evening I'll do a proof of concept to see if I can fix this and test it with my own Apple Developer Account.

PS. @goehmen I'm anbrown@pivotal.io - come up to the 5th floor and find me if you want to chat more about why I think good installers and first interactions are pretty important for building mind share outside of the typical IT sales pipeline.

**Comment by mdcarlson** [comment]
"brew install cloudfoundry-cli" works for me. Happy consumer here.


On Aug 28, 2015, at 12:43 AM, Andy Brown <notifications@github.com<mailto:notifications@github.com>> wrote:


Summary: My first ever interaction with cloud foundry was a negative one because the CLI installer on OS X is not properly signed. So... Bump.

Hi all, I'm a New-ish Pivotal employee @ Labs in SF. I signed up for a personal run.gopivotal.io account to learn about cloud foundry and maybe do some building in my free time. I have to say that my very first interaction with the product was trying to install the command line utility and FAILING to do so because of the security prompt on OS X.

This made me an unhappy user/customer as I grudgingly worked around the security prompt to install the CLI on OS X. After this first interaction with the product I did not feel like exploring more and went to bed. (A week later I'm playing with cloud foundry again and have successfully tracked down this existing bug)

For all our reference here is the Apple guide on signing packages for OS X:
https://developer.apple.com/library/mac/documentation/IDEs/Conceptual/AppDistributionGuide/DistributingApplicationsOutside/DistributingApplicationsOutside.html#//apple_ref/doc/uid/TP40012582-CH12-SW2

When I'm feeling industrious some other evening I'll do a proof of concept to see if I can fix this and test it with my own Apple Developer Account.

PS. @goehmen<https://github.com/goehmen> I'm anbrown@pivotal.io<mailto:anbrown@pivotal.io> - come up to the 5th floor and find me if you want to chat more about why I think good installers and first interactions are pretty important for building mind share outside of the typical IT sales pipeline.
